[Bug]: XmlPeek Result output item not properly escaped

### Issue Description

It looks like Result item from task XmlPeek is not properly escaped:

### Steps to Reproduce

Run `msbuild project.proj` with project.proj:

```xml
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="Build">
        <XmlPeek XmlInputPath="doc.xml" Query="//Key1/text()">
            <Output TaskParameter="Result" PropertyName="Key1Prop" />
            <Output TaskParameter="Result" ItemName="Key1Item" />
        </XmlPeek>
        <Message Text="Key1Prop=$(Key1Prop)" />
        <Message Text="Key1Item=%(Key1Item.Identity)" />

        <XmlPeek XmlInputPath="doc.xml" Query="//Key2/text()">
            <Output TaskParameter="Result" PropertyName="Key2Prop" />
            <Output TaskParameter="Result" ItemName="Key2Item" />
        </XmlPeek>
        <Message Text="Key2Prop=$(Key2Prop)" />
        <Message Text="Key2Item=%(Key2Item.Identity)" />

        <XmlPeek XmlInputPath="doc.xml" Query="//Key3/text()">
            <Output TaskParameter="Result" PropertyName="Key3Prop" />
            <Output TaskParameter="Result" ItemName="Key3Item" />
        </XmlPeek>
        <Message Text="Key3Prop=$(Key3Prop)" />
        <Message Text="Key3Item=%(Key3Item.Identity)" />
    </Target>
</Project>
```

and doc.xml:

```xml
<?xml version="1.0"?>
<Root>
    <Key1>abcdefg</Key1>
    <Key2>a$(d)fg</Key2>
    <Key3>a$(d.f)</Key3>
</Root>
```


### Expected Behavior

Prints text content from XML file. Properties and messages should agree:

```
  Key1Prop=abcdefg
  Key1Item=abcdefg
  Key2Prop=a$(d)fg
  Key2Item=afg
  Key3Prop=a$(d.f)
  Key3Item=a$(d.f)
 ```

### Actual Behavior

> project.proj(22,18): error MSB4184: The expression """.f" cannot be evaluated. Method 'System.String.f' not found.

### Analysis

_No response_

### Versions & Configurations

_No response_

--- a/src/Tasks.UnitTests/XmlPeek_Tests.cs
+++ b/src/Tasks.UnitTests/XmlPeek_Tests.cs
@@ -3,8 +3,10 @@
 
 using System;
 using System.IO;
+using System.Linq;
 
 using Microsoft.Build.Evaluation;
+using Microsoft.Build.Framework;
 using Microsoft.Build.Tasks;
 using Microsoft.Build.Utilities;
 
@@ -48,6 +50,13 @@ namespace Microsoft.Build.UnitTests
   <variable Type='String' Name='c'></variable>
   <method AccessModifier='public static' Name='GetVal' />
 </class>
+";
+        private readonly string _xmlFileRequiresEscaping = @"
+<Root>
+  <Key>abcdefg</Key>
+  <Key>a$(d)fg</Key>
+  <Key>a$(d.f)</Key>
+</Root>
 ";
 
         [Fact]
@@ -330,6 +339,24 @@ namespace Microsoft.Build.UnitTests
             project.Build().ShouldBeFalse();
             log.AssertLogContains("MSB4044");
             log.AssertLogContains("\"Query\"");
+        }
+
+        [Fact]
+        public void PeekEscapesCorrectly()
+        {
+            MockEngine engine = new MockEngine(true);
+            string xmlInputPath;
+            Prepare(_xmlFileRequiresEscaping, out xmlInputPath);
+
+            XmlPeek p = new XmlPeek();
+            p.BuildEngine = engine;
+
+            p.XmlInputPath = new TaskItem(xmlInputPath);
+            p.Query = "//Key/text()";
+
+            Assert.True(p.Execute());
+            Assert.Equal(["abcdefg", "a$(d)fg", "a$(d.f)"], p.Result.Select(x => x.ItemSpec));
+            Assert.Equal(["abcdefg", "a%24%28d%29fg", "a%24%28d.f%29"], p.Result.Cast<TaskItem>().Select(x => x.ToString()));
         }
 
         private void Prepare(string xmlFile, out string xmlInputPath)

--- a/src/Tasks/XmlPeek.cs
+++ b/src/Tasks/XmlPeek.cs
@@ -150,7 +150,7 @@ namespace Microsoft.Build.Tasks
             int i = 0;
             foreach (string item in peekValues)
             {
-                Result[i++] = new TaskItem(item);
+                Result[i++] = new TaskItem(EscapingUtilities.Escape(item));
 
                 // This can be logged a lot, so low importance
                 Log.LogMessageFromResources(MessageImportance.Low, "XmlPeek.Found", item);

